drivers/sensor/max30208/max30208.c

```diff
@@ -0,0 +1,234 @@
+	}
+
+	/* Read one sample */
+	error = max30208_readout_sample(dev);
+	if (error) {
+		return error;
+	}
+#endif
+
+	return 0;
+}
+
+static int max30208_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+	struct max30208_data *data = dev->data;
+
+	/* Convert raw value into temperature */
+	val->val1 = data->raw / MAX30208_ONE_DEGREE;
+	val->val2 = (data->raw % MAX30208_ONE_DEGREE) * MAX30208_LSB_e6;
+
+	return 0;
+}
+
+static const struct sensor_driver_api max30208_driver_api = {
+	.sample_fetch = max30208_sample_fetch,
+	.channel_get = max30208_channel_get,
+#ifdef CONFIG_MAX30208_TRIGGER
+	.trigger_set = max30208_trigger_set,
+	.attr_set = max30208_attr_set,
+#endif
+};
+
+static int max30208_init(const struct device *dev)
+{
+	struct max30208_data *data = dev->data;
+	const struct max30208_config *config = dev->config;
+
+	uint8_t part_id;
+	uint8_t sys_ctrl;
+
+	/* Get the I2C device */
+	data->i2c = device_get_binding(config->i2c_label);
+	if (!data->i2c) {
+		LOG_ERR("Could not find I2C device");
+		return -EINVAL;
+	}
+
+	/* Check the part id to make sure this is MAX30208 */
+	if (i2c_reg_read_byte(data->i2c, config->i2c_addr, MAX30208_REG_PART_ID, &part_id)) {
+		LOG_ERR("Could not get Part ID");
+		return -EIO;
+	}
+	if (part_id != MAX30208_PART_ID) {
+		LOG_ERR("Got Part ID 0x%02x, expected 0x%02x", part_id, MAX30208_PART_ID);
+		return -EIO;
+	}
+
+	/* Reset the sensor */
+	if (i2c_reg_write_byte(data->i2c, config->i2c_addr, MAX30208_REG_SYS_CTRL,
+			       MAX30208_RESET_MASK)) {
+		LOG_ERR("Failed to reset sensor!");
+		return -EIO;
+	}
+	/* Wait for reset to be cleared */
+	k_sleep(K_MSEC(10));
+	do {
+		if (i2c_reg_read_byte(data->i2c, config->i2c_addr, MAX30208_REG_SYS_CTRL,
+				      &sys_ctrl)) {
+			LOG_ERR("Couldn't read system control after reset");
+			return -EIO;
+		}
+	} while (sys_ctrl & MAX30208_RESET_MASK);
+
+	/* Write the FIFO configuration register */
+	if (i2c_reg_write_byte(data->i2c, config->i2c_addr, MAX30208_REG_FIFO_CFG1,
+			       config->fifo.fifo_a_full)) {
+		LOG_ERR("Failed to initialize FIFO config 1!");
+		return -EIO;
+	}
+	if (i2c_reg_write_byte(data->i2c, config->i2c_addr, MAX30208_REG_FIFO_CFG2,
+			       MAX30208_STC_TO_B(config->fifo.config2))) {
+		LOG_ERR("Failed to initialize FIFO config 2!");
+		return -EIO;
+	}
+
+	/* Write the GPIO configuration register */
+	if (i2c_reg_write_byte(data->i2c, config->i2c_addr, MAX30208_REG_GPIO_SETUP,
+			       MAX30208_STC_TO_B(config->gpio_setup))) {
+		LOG_ERR("Failed to initialize GPIO setup!");
+		return -EIO;
+	}
+
+#ifdef CONFIG_MAX30208_TRIGGER
+	if (max30208_init_interrupt(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupt!");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+static struct max30208_config max30208_config = {
+	.i2c_label = DT_INST_BUS_LABEL(0),
+	.i2c_addr = DT_INST_REG_ADDR(0),
```

Comment:
better use i2c_dt_spec

Reply:
I see. Dev began before the addition of `i2c_dt_spec`. How can I rebase my branch without losing my changes to a newer version?